examples/gcoap/gcoap_cli.c

```diff
@@ -26,6 +28,22 @@
 #include "net/gcoap.h"
 #include "od.h"
 #include "fmt.h"
+#ifdef MODULE_SOCK_DTLS
+#include "net/credman.h"
+
+#define SOCK_DTLS_GCOAP_TAG (10)
```

Comment:
More importantly, why define this value here? If it was defined in `gcoap.c` or `gcoap.h`, you would not need `gcoap_set_credential_tag()`, right? Is there any value to defining it in the application? Would gcoap ever need more than one credential? Do we need a distinction between a gcoap server credential and a client credential?

Reply:
Following this line of thinking, why not move all of the code in `gcoap_cli.c` into gcoap itself? gcoap provides CoAP as a service, so why should every application need to add this code? With this approach the application/user only needs to provide `credentials.c`.

Reply:
> More importantly, why define this value here? If it was defined in `gcoap.c` or `gcoap.h`, you would not need `gcoap_set_credential_tag()`, right? Is there any value to defining it in the application?

Yup, you're right. The credential tag is defined in the client to avoid reserving values for the tag used in gcoap. Other modules (e.g. LWM2M) might also hard-code the tag used for the DTLS sock and then there will be a need to track the available tag somewhere to avoid collisions.

> Would gcoap ever need more than one credential?

Yes. gcoap might be talking to other secure coap clients/servers that use different credentials.

> Do we need a distinction between a gcoap server credential and a client credential?

No.

> Following this line of thinking, why not move all of the code from this PR in `gcoap_cli.c` into gcoap itself? gcoap provides CoAP as a service, so why should every application need to add this code? With this approach, the application/user only needs to provide `credentials.c`.

The credentials are added in `gcoap_cli.c` because we do not want to make any assumption or limit what the name of the variables used for the credentials should be. 

In general, the design of coupling DTLS sock with credman is to avoid limiting what the name of the credentials should be or where it can be found. This is left to the user to define.